hrms/hr/report/shift_attendance/shift_attendance.py

```diff
@@ -0,0 +1,166 @@
+# Copyright (c) 2023, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe import _
+from frappe.utils import flt
+
+
+def execute(filters=None):
+	columns = get_columns()
+	data = get_data(filters)
+	return columns, data
+
+
+def get_columns():
+	return [
+		{
```

Comment:
Adjust column widths as per the data

<img width="1335" alt="image" src="https://github.com/frappe/hrms/assets/24353136/a42fb186-9695-4f13-9ea0-f2d222a2594e">

- Employee Name is sort of redundant since reports have formatters for employee link fields (Name: Employee Name). You can remove this field
- Late entry, Early Exit can take up lesser space
- Since these columns are anyway going to the right, adjust column widths to show the complete column header
<img width="552" alt="image" src="https://github.com/frappe/hrms/assets/24353136/dddfe065-1e84-417a-b862-77c4c0702240">

Reply:
hi brother. absent not showing in the report.